accounts/abi/bind/base_test.go

```diff
@@ -0,0 +1,29 @@
+package bind
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+func TestUnkownEventName(t *testing.T) {
+	errorPrefix := "abi: event"
+	c := &BoundContract{}
+	c.abi.Events = make(map[string]abi.Event)
+	c.abi.Events["event1"] = abi.Event{}
+	if _, _, err := c.FilterLogs(nil, "event_not_exist"); err == nil || !strings.HasPrefix(err.Error(), errorPrefix) {
+		t.Fatal("should report error if event not found")
```

Comment:
Grammar check: should report AN error if THE event IS not found

Also, given your check if the `if` above, an error could be found that isn't the one that you expect. So you can either change the sentence to "should report an error that the event is unknown" or break the check into two: one that fails if the error is `nil`, and one that fails if the error isn't the correct one.